packages/flutter/lib/src/scheduler/ticker.dart

```diff
@@ -12,80 +15,233 @@ import 'binding.dart';
 /// at the time of the callback being called.
 typedef void TickerCallback(Duration elapsed);
 
+/// An interface implemented by classes that can vend [Ticker] objects.
+abstract class TickerProvider {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const TickerProvider();
+
+  /// Creates a ticker with the given callback.
+  ///
+  /// The kind of ticker provided depends on the kind of ticker provider.
+  Ticker createTicker(TickerCallback onTick);
+
+  /// A [TickerProvider] that creates [Ticker] objects that will call `onTick`
+  /// once per frame while active.
+  static const TickerProvider schedulerTickerProvider = const _DefaultTickerProvider();
```

Comment:
This field seems like a bug factory.  It's easy to use and your code is only wrong in that it burns battery.  Maybe move it to `flutter_test`, which would service almost all of the clients.